book_service/book_service.go

```diff
@@ -78,7 +78,12 @@ func (b *BookService) DeleteBook(ctx context.Context, id string, username string
 }
 
 func (b *BookService) Search(title string, author string, username string, routeName string, priceRange string) ([]models.Book, error) {
-	res, err := b.dbHandler.Search(title, author, priceRange)
+	searchParams := make(map[string]string)
+	searchParams["title"] = title
+	searchParams["author"] = author
+	searchParams["price_range"] = priceRange
+
+	res, err := b.dbHandler.Search(searchParams)
```

Comment:
Same for the `BookService` - `Search` function

Reply:
If you mean to change the searchParms to fields, then it's done :)